packages/ai/src/PromptInput.ts

```diff
@@ -0,0 +1,285 @@
+
+	/**
+	 * Defines the label of the input field.
+	 *
+	 * @default ""
+	 * @since 2.0.0-rc.1
+	 * @public
+	 */
+	@property()
+	label!: string;
+
+	/**
+	 * Defines whether the clear icon of the input will be shown.
+	 * @default false
+	 * @public
+	 * @since 2.0.0-rc.1
+	 */
+	@property({ type: Boolean })
+	showClearIcon!: boolean;
+
+	/**
+	 * Determines whether the characters exceeding the maximum allowed character count are visible
+	 * in the component.
+	 *
+	 * If set to `false`, the user is not allowed to enter more characters than what is set in the
+	 * `maxlength` property.
+	 * If set to `true` the characters exceeding the `maxlength` value are selected on
+	 * paste and the counter below the component displays their number.
+	 * @default false
+	 * @public
+	 * @since 2.0.0-rc.1
+	 */
+	@property({ type: Boolean })
+	showExceededText!: boolean;
+
+	/**
+	 * Defines whether the component is in disabled state.
+	 *
+	 * **Note:** A disabled component is completely noninteractive.
+	 * @default false
+	 * @public
+	 * @since 2.0.0-rc.1
+	 */
+	@property({ type: Boolean })
+	disabled!: boolean;
+
+	/**
+	 * Defines whether the component is read-only.
+	 *
+	 * **Note:** A read-only component is not editable,
+	 * but still provides visual feedback upon user interaction.
+	 * @default false
+	 * @public
+	 * @since 2.0.0-rc.1
+	 */
+	@property({ type: Boolean })
+	readonly!: boolean;
+
+	/**
+	 * Sets the maximum number of characters available in the input field.
+	 *
+	 * @default undefined
+	 * @since 2.0.0-rc.1
+	 * @public
+	 */
+	@property({ validator: Integer })
+	maxlength?: number;
+
+	/**
+	 * Defines the value state of the component.
+	 * @default "None"
+	 * @since 2.0.0-rc.1
+	 * @public
+	 */
+	@property({ type: ValueState, defaultValue: ValueState.None })
+	valueState!: `${ValueState}`;
+
+	@property({ validator: Integer })
+	_charactersLeft?: number;
+
+	/**
+	 * Defines the value state message that will be displayed as pop up under the component.
+	 * The value state message slot should contain only one root element.
+	 *
+	 * **Note:** If not specified, a default text (in the respective language) will be displayed.
+	 *
+	 * **Note:** The `valueStateMessage` would be displayed,
+	 * when the component is in `Information`, `Critical` or `Negative` value state.
+	 *
+	 * @since 2.0.0-rc.1
+	 * @public
+	 */
+	@slot({
+		type: HTMLElement,
+		invalidateOnChildChange: true,
+	})
+	valueStateMessage!: Array<HTMLElement>;
+
+	_isEnter?: boolean;
+	_exceededTextProps!: ExceededText;
+
+	static i18nBundle: I18nBundle;
+
+	static async onDefine() {
+		PromptInput.i18nBundle = await getI18nBundle("@ui5/webcomponents");
+	}
+
+	constructor() {
+		super();
+		this._isEnter = false;
+	}
+
+	_onkeydown(e: KeyboardEvent) {
+		if (isEnter(e)) {
+			this._isEnter = true;
+		}
+	}
+
+	_onInnerInput(e: CustomEvent<InputEventDetail>) {
+		this.value = (e.target as Input).value;
+
+		this.fireEvent("input");
+	}
+
+	_onInnerChange() {
```

Comment:
isn't it more correct to listen for keydown event and to check if `Enter` key is pressed instead of this change event? Like we do proxy for the change event simplifying this to:
```
	_onInnerChange() {
		this.fireEvent("change");
	}
```

and changing the `_onkeydown` to:

```ts
	_onkeydown(e: KeyboardEvent) {
		if (isEnter(e)) {
			this.fireEvent("submit");
		}
	}
```

then this _isEnter key becomes redundant